### ISSUE TITLE
waste of time

### ISSUE THREAD

**Issue by avinashkavi447** [hashicorp/vscode-terraform#249]
Issue Type: <b>Bug</b>

extenstions are not giving suggestions

Extension version: 1.4.0
VS Code version: Code 1.40.2 (f359dd69833dd8800b54d458f6d37ab7c78df520, 2019-11-25T14:54:45.096Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i3-3227U CPU @ 1.90GHz (4 x 1896)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.88GB (3.40GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by endzyme** [comment]
This is not a very helpful issue to someone who is supposed to resolve your experience. I would suggest looking at some other issues in this repo and documenting what you tried and what you expected to happen. This is open source and people are only able to help if they are given a fair chance to make things better.

**Comment by Bozydarek** [comment]
I believe that this is connected with this https://github.com/mauve/vscode-terraform/issues/211.
It's look like current version of LSP (0.0.10) use some very inefficient way to tokenize and parse terraform files. This is why it stuck (with 100% cpu usage) on "big" files (>50 lines, multiple resources) and can't prepare suggestions for user.

**Comment by juliosueiras** [comment]
@Bozydarek can you check again, since that CPU usage issue is fixed(was due to a recursion bug)

**Comment by Bozydarek** [comment]
![obraz](https://user-images.githubusercontent.com/16276635/75044298-af054180-54c1-11ea-80f3-a18d80751350.png)

I don't see any different. Is there any newer version then this:
```
$ ./terraform-lsp --version
v0.0.10, commit: 058c36f7fe4b68a7335cc63e066deebe3a3b238c, build on: 2020-02-17T14:32:20Z
```

**Comment by juliosueiras** [comment]
@Bozydarek A) which os? and B) can you contact me in https://gitter.im/terraform-lsp/community (since then it will be easier for me to ask questions regarding the situation) ?

**Comment by paultyng** [comment]
We just released [v2.0.0-rc.1](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) of the extension. The main features include:

* Added syntax support for 0.12
* Added [terraform-ls](https://github.com/hashicorp/terraform-ls) usage by default (currently on 0.3.0, which offers basic provider code completion)

You can find additional information and specifics in the [release notes](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) and [CHANGELOG](https://github.com/hashicorp/vscode-terraform/blob/master/CHANGELOG.md).

With this release we expect that many of the prior issues and PRs are no longer relevant or have been addressed, and are therefore being closed. If you feel the action taken on an issue or PR is in error, please comment as such and we can figure out the appropriate way to address it.

We plan to add the final 2.0.0 release to the marketplace soon, but are actively seeking your feedback now on the release candidates. You can download the .vsix from the releases page and [manually install it in VS Code](https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix) to try it out.
